libutils/file_lib.c

```diff
@@ -1031,8 +1031,38 @@ FILE *safe_fopen_create_perms(
     {
         return NULL;
     }
-    FILE *ret = fdopen(fd, mode);
-    if (!ret)
+
+    /* These are the modes supported by fdopen() */
+    static const char *modes[] =
+    {
+        "a+b", "ab+", "a+",
+        "w+b", "wb+", "w+",
+        "r+b", "rb+", "r+",
+        "ab", "a",
+        "wb", "w",
+        "rb", "r",
+        NULL,
+    };
+
+    /* Try to use only supported mode strings for fdopen() */
```

Comment:
I think a better approach would be to build a valid `char fdopen_mode[4]` string in the `switch` above, only adding the letters that are supported.

Reply:
Yup, that sounds better, I will try :wink: